assets/js/base/utils/legacy-events.js

```diff
@@ -1,19 +1,51 @@
 const Event = window.Event || null;
 
-// This is a hack to trigger cart updates till we migrate to block based cart
-// that relies on the store, see
-// https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1247
-export const triggerFragmentRefresh = () => {
+/**
+ * Wrapper function to dispatch an event so it's compatible with IE11.
+ *
+ * @param {string}  name       Name of the event to dispatch.
+ * @param {boolean} bubbles    Whether the event bubbles.
+ * @param {boolean} cancelable Whether the event is cancelable.
+ */
+const dispatchEvent = ( name, bubbles = false, cancelable = false ) => {
 	// In IE, Event is an object and can't be instantiated with `new Event()`.
 	if ( typeof Event === 'function' ) {
-		const event = new Event( 'wc_fragment_refresh', {
-			bubbles: true,
-			cancelable: true,
+		const event = new Event( name, {
+			bubbles,
+			cancelable,
 		} );
 		document.body.dispatchEvent( event );
 	} else {
 		const event = document.createEvent( 'Event' );
-		event.initEvent( 'wc_fragment_refresh', true, true );
+		event.initEvent( name, bubbles, cancelable );
 		document.body.dispatchEvent( event );
 	}
 };
+
+// This is a hack to trigger cart updates till we migrate to block based cart
+// that relies on the store, see
+// https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1247
+export const triggerFragmentRefresh = () => {
+	dispatchEvent( 'wc_fragment_refresh', true, true );
+};
+
+/**
+ * Function that listens to a jQuery event and dispatches a native JS event.
+ * Useful to convert WC Core events into events that can be read by blocks.
+ *
+ * @param {string}  jQueryEventName Name of the jQuery event to listen to.
+ * @param {string}  nativeEventName Name of the native event to dispatch.
+ * @param {boolean} bubbles         Whether the event bubbles.
+ * @param {boolean} cancelable      Whether the event is cancelable.
+ */
+export const translateJQueryEventToNative = (
+	jQueryEventName,
+	nativeEventName,
+	bubbles = false,
+	cancelable = false
+) => {
+	// @ts-ignore jQuery is window global
```

Comment:
Typical pattern for comments on linting blocks is something like this:

```suggestion
	// @ts-ignore -- jQuery is window global
```